Bottom: 443e25aee8473476be4f31b83fb589c9ca9b5365
Top:    5de15b001b6271de3a6db9b3b1cf7f98bb75cb4f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-03 23:13:27 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index cb40e4d..86235d2 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -307,6 +307,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
